### PR TITLE
[7.x] Use external ascii package

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Back in the day, Laravel used Stringy for ascii conversion, however that package has gone out of maintenance, and Laravel doesn't use it any more. A fork of Stringy has prevailed, and become popular, and I have asked the author to extract the ascii part of the package for to a small package without any dependencies. The current maintainer has over 1 million package downloads on packagist, and cares about code quality, so we're in good hands.

NB There is a small change compared to the implementation in Laravel 6, in that Bulgarian translation has been corrected. The "ь" and "Ь" characters don't make sense on their own - I've been advised as to the correct implementation by a native speaker: https://twitter.com/suckup_de/status/1170689784249032704. We also have it from a native German speaker that the correct capitalization of `Ä` is `Ae` in ascii, such as when `Ä` is used as the first letter of a word (https://github.com/voku/portable-ascii/issues/2#issuecomment-529994426).

---

TLDR:

1. Maintenance of Str::ascii is a pain, and we should consider moving to a well used 3rd party library.
2. I have selected one close to what Laravel's implementation was based on.
3. The new library handles a lot of language specific stuff better than Laravel's implementation, but one could consider this breaking, so I've targetted Laravel 7.

### CODE REVIEW

**Comment by driesvints** [comment]
@GrahamCampbell this is looking good. Nice job 👍

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -207,6 +198,17 @@ public static function is($pattern, $value)
         return false;
     }
 
+    /**
+     * Transliterate a UTF-8 value to ASCII.
```
Comment (derekmd) [inline src/Illuminate/Support/Str.php]: It looks like this sentence was copy and pasted from `Str::ascii()`'s docblock?

`ASCII::is_ascii()`'s comment is:

```
/**
 * Checks if a string is 7 bit ASCII.
 *
```

But Laravel docblock grammar for predicates is usually "Determine if...".

```
/**
 * Determine if a given string is 7 bit ASCII.
 *
```

Comment (GrahamCampbell) [reply src/Illuminate/Support/Str.php]: Ah, yes. Good catch! Yeh, I copied the whole function, but forgot to update the doc.

**Comment by barryvdh** [comment]
What about the new Symfony String component? That handles ASCII and other cases. https://github.com/symfony/symfony/pull/33553

**Comment by driesvints** [comment]
Ah nice. I think that was only added after this PR. Still loads of time until 7.0 so we can still get that in.

**Comment by barryvdh** [comment]
Not really sure what branch it is though, because it's only in master and probably experimental, so maybe even for 8.0

**Comment by GrahamCampbell** [comment]
Symfony's string component is not planning on providing an interface that lets you choose a language.

**Comment by GrahamCampbell** [comment]
Also, Symfony's string component will be using the same ascii package as us. ;P

**Comment by GrahamCampbell** [comment]
Basically, we should not change our implementation.
